src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -15,11 +15,43 @@ internal class MsQuicApi : IDisposable
 
         private readonly IntPtr _registrationContext;
 
+        public static bool IsQuicSupported { get; } = GetIsQuicSupported();
+
+        private static bool GetIsQuicSupported()
+        {
+            // TODO: check using MsQuicOpen return value; but there doesn't seem to be a consistent error code for this yet.
+            OperatingSystem ver = Environment.OSVersion;
+
+            if (ver.Platform != PlatformID.Win32NT || ver.Version >= new Version(10, 0, 19041, 0))
+            {
+                try
+                {
+                    new MsQuicApi().Dispose();
```

Comment:
Instead of newing up a MsQuicApi, can you just create a new property (ex: IsLoaded) that would be set in the private constructor and check `Api.IsLoaded`

Reply:
Good idea.